qt/widgets/common/src/FitScriptGeneratorModel.cpp

```diff
@@ -157,148 +108,415 @@ void FitScriptGeneratorModel::addWorkspaceDomain(
                                 std::to_string(workspaceIndex.value) +
                                 ")' domain already exists.");
 
-  addWorkspaceDomain(nextAvailablePrefix(), workspaceName, workspaceIndex,
-                     startX, endX);
+  m_fitDomains.emplace_back(
+      std::make_unique<FitDomain>(workspaceName, workspaceIndex, startX, endX));
 }
 
-void FitScriptGeneratorModel::addWorkspaceDomain(
-    std::string const &prefix, std::string const &workspaceName,
-    WorkspaceIndex workspaceIndex, double startX, double endX) {
-  addEmptyCompositeAtPrefix(prefix);
-  m_fitDomains.emplace_back(
-      FitDomain(prefix, workspaceName, workspaceIndex, startX, endX));
+bool FitScriptGeneratorModel::hasWorkspaceDomain(
+    std::string const &workspaceName, WorkspaceIndex workspaceIndex) const {
+  return findWorkspaceDomain(workspaceName, workspaceIndex) !=
+         m_fitDomains.cend();
 }
 
-std::size_t
+FitDomainIndex
```

Comment:
I have just had a thought about this. Where is it defined? Is it worth having it as a custom item? @Harrietbrown do you have any thoughts on `FitDomainIndex` and how its used in the indirect code base?

Reply:
It is defined in common widgets in `IndexTypes.h`. It is currently being used widely in indirect (I think Roman introduced it when he did simultaneous fitting), and so this is already good 👍

Reply:
I know @Harrietbrown found some of the custom indexes confusing. I'm not sure if this is one of them

Reply:
I have found these indexes quite useful so far